parkings/tests/api/operator/test_permit.py

```diff
@@ -0,0 +1,259 @@
+
+    assert response.status_code == HTTP_204_NO_CONTENT
+    assert Permit.objects.count() == 5
+    assert not Permit.objects.filter(id=operator1_permit.id).exists()
+
+
+def test_operator_and_enforcers_cannot_see_each_others_permit(
+    operator_api_client, permit_series_factory, operator, staff_api_client,
+    staff_user, permit_factory, enforcement_domain, enforcer_factory
+):
+    reg_number = 'ABC-123'
+    permit_subject = generate_subjects()
+    permit_subject[0].update(registration_number=reg_number)
+
+    operator_permitseries = permit_series_factory(owner=operator.user)
+    enforcer_permitseries = permit_series_factory(owner=staff_user)
+
+    enforcer_permit_list = [
+        permit_factory(subjects=permit_subject, series=enforcer_permitseries)
+        for _
+        in range(3)
+    ]
+    enforcer_default_domain = enforcer_permit_list[0].domain
+    enforcer_factory(user=staff_user, enforced_domain=enforcer_default_domain)
+
+    operator_permit_domain = [enforcement_domain, enforcer_default_domain]
+    operator_permit_list = [
+        permit_factory(subjects=permit_subject, domain=domain, series=operator_permitseries)
+        for domain
+        in operator_permit_domain
+        ]
+
+    #  Operator should see only operator_permit_list
+    response = operator_api_client.get(list_url)
+    json_response = response.json()
+
+    assert response.status_code == HTTP_200_OK
+    assert json_response['count'] == 2
+    assert set([permit['id'] for permit in json_response['results']]) == set(
+        [permit.id for permit in operator_permit_list]
+    )
+
+    enforcement_permit_url = reverse('enforcement:v1:permit-list')
+
+    #  Enforcer should see only enforcer_permit_list
+    response = staff_api_client.get(enforcement_permit_url)
+    json_response = response.json()
+
+    assert response.status_code == HTTP_200_OK
+    assert json_response['count'] == 3
+    assert set([permit['id'] for permit in json_response['results']]) == set(
+        [permit.id for permit in enforcer_permit_list]
+    )
+
+
+def test_operator_cannot_create_permit_on_permitareas_where_she_is_not_permitted(
+    operator_api_client, permit_series_factory, enforcer_factory, operator
+):
+    area = create_area_geom()
+    permit_series = permit_series_factory(active=True)
+    operator_enforcer = enforcer_factory(user=operator.user)
+    permit_data = _get_permit_data(permit_series)
+    permit_data.update(domain=operator_enforcer.enforced_domain.code)
+
+    PermitArea.objects.create(
+        name='Kamppi',
+        identifier='A',
+        geom=area,
+        permitted_user=operator.user
+    )  # PermitArea with different domain compared to permit_data
+
+    for each_area in permit_data['areas']:
+        each_area.update(area='A')
+
+    response = operator_api_client.post(list_url, data=permit_data)
+
+    assert response.json()['non_field_errors'] == [
+        'You can\'t create permit whose domain is different than areas\'s domain.'
```

Comment:
Hmm.. the permit area is looked up by the (domain, area identifier) pair, so it's not possible by definition to create a permit which references an area in a different domain. Either the pair (HKI, A) is a valid key for a permit area or it's not.